### PR TITLE
More Colors, Bug Fixes and Transitions

### PR DESCRIPTION
**Pull request by oltmannsdaniel**
- More Colors (Buttons, Inputs)
- Transitions (Button hover, Input Focus)
- Button disabled Bug fix
- Textarea only vertical resize (Why is more than 100% width possible?)

### CODE REVIEW

**Comment by enlore** [comment]
This looks cool. Give a some time to pull it down and mess with it. I'll circle back again on Monday.

**Comment by oltmannsdaniel** [comment]
@enlore Thanks. I will add some more functions to use colors soon. Like Text Colors, Row Colors, etc.

File: src/core/_buttons.css [hunk 1]
```diff
@@ -19,6 +19,11 @@ input[type="button"] {
   border-radius: var(--border-radius);
   border: var(--button-border);
   cursor: pointer;
+  -webkit-transition: border 0.2s;
```
Comment (gld1982ltd) [inline src/core/_buttons.css]: All prefixing is handled by PostCSS with Autoprefixer. Could you remove these from the src files, please, as they will be in the dist files. If some necessary prefixes are missing, you can report a bug at [Autoprefixer's issue tracker](https://github.com/postcss/autoprefixer/issues).

**Comment by gld1982ltd** [comment]
Maybe alerts should be placed into bones...

**Comment by oltmannsdaniel** [comment]
@gld1982ltd I will keep this in the Pull Request. In bones we can add a Navbar, Tab View and Checkboxes as well. All pure CSS and as less as possible lines.

File: src/core/_buttons.css (hunk 1 above)
Comment (oltmannsdaniel) [inline src/core/_buttons.css]: But I want to support webkit,moz,ms and o! How I can change this? In the build only webkit has been added.

Comment (gld1982ltd) [inline src/core/_buttons.css]: Only webkit I'd needed. Someone correct me if I'm wrong...  http://caniuse.com/#feat=css-transitions

**Comment by gld1982ltd** [comment]
@enlore I'm waiting on your results. I'll check back in later.

File: src/core/_buttons.css (hunk 1 above)
Comment (oltmannsdaniel) [inline src/core/_buttons.css]: Only Opera (-o-) is useless. In my view.

Comment (enlore) [inline src/core/_buttons.css]: The build process uses Autoprefixer. Autoprefixer uses data from http://caniuse.com/ to determine which prefixes are needed based on a set of "target" browsers. 

The default set of target browsers is defined by the Browserlist module as `> 1%, last 2 versions, Firefox ESR`, which roughly translates to "any browser with greater that 1% market share, the newest two versions of all browsers, and the latest Firefox ESR version".

See https://github.com/ai/browserslist for a breakdown of what that string means in more detail.

Manually adding prefixes for browsers is therefore (in theory, anyway) unnecessary.

To customize Skeleton's browser support... Well we probably should make a thing that does that. @gld1982ltd, how do you feel about adding something like a config file to this project that your build process module picks up and uses to tell Autoprefixer what to do?

**Comment by enlore** [comment]
Most of this looks good. I like the colors, the transitions on the form elements, and the `texarea` fix. The `alert` classes are nice, too, for creating content that really pops out. Good stuff.

Tables are a little fiddly. I'm looking for them to scroll horizontally at small screen sizes. Currently the cells stack:

image

Could you remove the `display: block` from `td` and `th` elements in the `_tables.css` module?

For extra credit, you could make a `u-table-responsive` class, too. We could use this class to wrap tables in a div to accomplish horizontal scrolling tables, a la [responsive tables](http://getbootstrap.com/css/#tables-responsive) in Bootstrap.  If not, I'll put that together after I merge in your work. Either way, here's the intended result:

image

image

Also, we have utility classes that do `text-align: right` and `text-align: center` already. Having a `text-left` class isn't a bad idea though, so could you stick with our established pattern and just make a `u-text-left` class in `core/_utils.css`? 

Lastly, drop the vendor prefixes and let Autoprefixer do its thing. I think adding a feature to the build process to target specific browsers wouldn't be a bad idea, but that's out of scope for this commit.

Thanks!

**Comment by oltmannsdaniel** [comment]
@enlore I will remove the text-center etc. today. And it would be helpful if you edit the table.css aftee the merge. Also we can work on an new work Process. Maybe with a Gulpfile.

It works pretty good and is used also in Other Projects. Easy to edit. Easy to build. Easy extandable.

Hopfully we move to Gulp.

File: dist/index.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
   <meta charset="utf-8">
   <title>Skeleton Framework Test File</title>
   <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link href="//fonts.googleapis.com/css?family=Raleway:400,300,600" rel="stylesheet" type="text/css">
+  <link href="http://fonts.googleapis.com/css?family=Raleway:400,300,600" rel="stylesheet" type="text/css">
```
Comment (gld1982ltd) [inline dist/index.html]: Why http? https://developers.google.com/fonts/docs/getting_started#a_quick_example

File: src/core/_buttons.css (hunk 1 above)
Comment (gld1982ltd) [inline src/core/_buttons.css]: The postCSS options can be configured here https://github.com/gld1982ltd/skeleton-postcss as we agreed to purge this repo of build libs. I am actually leaning toward postcss-cli as our runner allowing everything to be configured and ran from the same package.json.

File: dist/index.html (hunk 2 above)
Comment (oltmannsdaniel) [inline dist/index.html]: I added http because the live-serve doesn't work on my PC. So I load it with file:// and the font cannot load over file:// .

File: src/core/_buttons.css (hunk 1 above)
Comment (oltmannsdaniel) [inline src/core/_buttons.css]: Gulp is also a good way to build tghe CSS. There are many repos that use Gulp and there are also many packages to extend this Process.

File: dist/index.html (hunk 2 above)
Comment (gld1982ltd) [inline dist/index.html]: Are you running `npm start` to fire up the server? If live-serve isn't working please submit a new issue. Any build/development issues can be resolved in a different commit.

**Comment by gld1982ltd** [comment]
@oltmannsdaniel We were using gulp, but for this project it is a little overboard. PostCSS is perectly capable of doing this without gulp. As I stated above, any build/development issues can be resolved in a different commit.

**Comment by gld1982ltd** [comment]
@enlore Can you go ahead and merge this with the table fixes you mentioned? I'll start fixing up the build system later today, and should be done with it by tonight to submit a PR.

**Comment by enlore** [comment]
Will do.

On Wed, May 11, 2016, 7:41 AM Gary DeLaney <notifications@github.com> wrote:

> @enlore <https://github.com/enlore> Can you go ahead and merge this with
> the table fixes you mentioned? I'll start fixing up the build system later
> today, and should be done with it by tonight to submit a PR.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/skeleton-framework/skeleton-framework/pull/46#issuecomment-218447341>
>
-- 
Nick Lorenson
Dev @ HCA
m | 931 241 1014
nick@elcreativegroup.com
n.e.lorenson@gmail.com

Legal stuff: for our mutual protection, anything we send back and forth via
email is to be considered confidential unless stated otherwise.
